### PR TITLE
Fixing cli test failure in CI

### PR DESCRIPTION
**Pull request by amoghrajesh**
Fixing https://github.com/apache/airflow/actions/runs/12171824313/job/33949649869

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by amoghrajesh** [comment]
This PR might not be needed

File: tests/cli/test_cli_parser.py [hunk 1]
```diff
@@ -423,7 +423,7 @@ def test_invalid_choice_raises_for_export_format_in_db_export_archived_command(
         assert error_msg == (
             "\nairflow db export-archived command error: argument "
             f"--export-format: invalid choice: '{export_format}' "
```
Comment (gopidesupavan) [inline tests/cli/test_cli_parser.py]:
```suggestion
        assert error_msg.replace("'", "") == (
            "\nairflow db export-archived command error: argument "
            f"--export-format: invalid choice: {export_format}"
```

As Jarek, mentioned in slack this is happening for 3.12, so may be we can do like this then it should work for all versions, know this is bit ugly but dont have any other option :)

Comment (amoghrajesh) [reply tests/cli/test_cli_parser.py]: I dont love this idea but I do not have a better one too.
I am thinking instead, we just add  a check for PY3.12+ and check?

@potiuk @gopidesupavan wdyt?

Comment (potiuk) [reply tests/cli/test_cli_parser.py]: Maybe:

```
assert "airflow db export-archived command error: argument --export-format: invalid choice" in error_msg
```

We do not have to test for exact equality

Comment (amoghrajesh) [reply tests/cli/test_cli_parser.py]: Yeah good idea, thanks.

**Review by potiuk — APPROVED** [review]

**Pull request merged**